plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp

```diff
@@ -241,16 +241,16 @@ inline bool ProxyShape::TransformReference::decRef(const TransformReason reason)
             m_required);
     switch (reason) {
     case kSelection:
-        assert(m_selected);
```

Comment:
This function needs to be able to exit in Debug, the returned value is tested properly and this is why it is working in release. Also making sure not to set negative values.

Reply:
Not sure I understand this one completely. It feels like the intention was to never get into a situation where increment and decrement pattern is not respected.

Reply:
@fabal Can you review and approve this change to the AL plugin?

Reply:
The starting point of the case that was popping the assert: maya-usd\plugin\al\plugin\AL_USDMayaTestPlugin\AL\usdmaya\nodes\test_ProxyShape.cpp (1007) :
// construct a chain of transform nodes
            proxy->removeUsdTransforms(
                kneePrim, modifier3, AL::usdmaya::nodes::ProxyShape::kRequested);
            EXPECT_EQ(MStatus(MS::kSuccess), modifier3.doIt());